tools/GdkTestRunner/Modules/ModuleLibrary.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json.Linq;
+using NLog;
+
+namespace GdkTestRunner.Modules
+{
+    public static class ModuleLibrary
+    {
+        public static List<string> LoadedModuleIdentifiers => moduleTypeLibrary.Keys.ToList();
+
+        private static readonly Dictionary<string, Type> moduleTypeLibrary = new Dictionary<string, Type>();
+        private static readonly Dictionary<string, string> moduleDocumentation = new Dictionary<string, string>();
+
+        static ModuleLibrary()
+        {
+            var types = AppDomain.CurrentDomain.GetAssemblies().SelectMany(assembly => assembly.GetTypes());
+
+            var moduleTypes = types.Where(type => typeof(BaseModule).IsAssignableFrom(type) && !type.IsAbstract);
+
+            foreach (var moduleType in moduleTypes)
+            {
+                var identifierAttribute = Attribute.GetCustomAttribute(moduleType,
+                    typeof(TestModuleIdentifierAttribute)) as TestModuleIdentifierAttribute;
+
+                if (identifierAttribute == null)
+                {
+                    LogManager.GetCurrentClassLogger().Error($"Encountered subclass of {typeof(BaseModule).FullName} " +
+                        $"without {typeof(TestModuleIdentifierAttribute).FullName}");
+                    continue;
+                }
+
+                moduleTypeLibrary[identifierAttribute.Identifier] = moduleType;
+                moduleDocumentation[identifierAttribute.Identifier] = identifierAttribute.Docs;
+            }
+        }
+
+        public static BaseModule CreateModuleFromType(string type, JToken jsonContext, GdkTestRunnerOptions options)
+        {
+            if (!moduleTypeLibrary.TryGetValue(type, out var moduleType))
+            {
+                throw new ArgumentException($"Cannot find a module with JSON identifier: {type}");
+            }
+
+            return (BaseModule) Activator.CreateInstance(moduleType, options, jsonContext);
+        }
+
+        public static string GetModuleDocumentation(string identifier)
+        {
+            if (!moduleDocumentation.TryGetValue(identifier, out var docs))
+            {
+                throw new ArgumentException($"Could not find a module with identifier: {identifier}");
```

Comment:
very small nit: one time you wrote `Cannot ... ` and here you use `Could not ... ` ;)